exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -134,20 +134,68 @@ func (prwe *PRWExporter) PushMetrics(ctx context.Context, md pdata.Metrics) erro
 
 					// handle individual metric based on type
 					switch metric.DataType() {
-					case pdata.MetricDataTypeDoubleSum, pdata.MetricDataTypeIntSum, pdata.MetricDataTypeDoubleGauge, pdata.MetricDataTypeIntGauge:
-						if err := prwe.handleScalarMetric(tsMap, resource, metric); err != nil {
+					case pdata.MetricDataTypeDoubleGauge:
+						dataPoints := metric.DoubleGauge().DataPoints()
+						if dataPoints.Len() == 0 {
 							dropped++
-							errs = append(errs, consumererror.Permanent(err))
+							errs = append(errs, consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name())))
 						}
-					case pdata.MetricDataTypeHistogram, pdata.MetricDataTypeIntHistogram:
-						if err := prwe.handleHistogramMetric(tsMap, resource, metric); err != nil {
+						for x := 0; x < dataPoints.Len(); x++ {
+							addSingleDoubleDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
+						}
+					case pdata.MetricDataTypeIntGauge:
+						dataPoints := metric.IntGauge().DataPoints()
+						if dataPoints.Len() == 0 {
+							dropped++
+							errs = append(errs, consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name())))
+						}
+						for x := 0; x < dataPoints.Len(); x++ {
+							addSingleIntDataPoint(dataPoints.At(x), resource, metric, prwe.namespace, tsMap, prwe.externalLabels)
+						}
+					case pdata.MetricDataTypeDoubleSum:
+						dataPoints := metric.DoubleSum().DataPoints()
+						if dataPoints.Len() == 0 {
 							dropped++
-							errs = append(errs, consumererror.Permanent(err))
+							errs = append(errs, consumererror.Permanent(fmt.Errorf("empty data points. %s is dropped", metric.Name())))
```

Comment:
In a separate PR we should discuss about this, I don't think this should be an error.